vowpalwabbit/cb_explore.cc

```diff
@@ -255,15 +256,13 @@ void print_update_cb_explore(vw& all, bool is_test, example& ec, std::stringstre
   }
 }
 
-void output_example(vw& all, cb_explore& data, example& ec, CB::label& ld)
+void generic_output_example(vw& all, example& ec, CB::label& ld)
 {
   float loss = 0.;
+  auto known_cost = get_observed_cost(ld);
 
-  cb_to_cs& c = data.cbcs;
-
-  if ((c.known_cost = get_observed_cost(ld)) != nullptr)
-    for (uint32_t i = 0; i < ec.pred.a_s.size(); i++)
-      loss += get_cost_estimate(c.known_cost, c.pred_scores, i + 1) * ec.pred.a_s[i].score;
+  if (known_cost != nullptr)
+    loss = CB_ALGS::get_cost_estimate(known_cost, known_cost->action) * ec.pred.a_s[known_cost->action - 1].score;
```

Comment:
this loss change changes legacy cb_explore, @pmineiro that ok?

Reply:
Hmm.  Now we are computing an IPS estimate.  We'll need to dig into that other overload of `get_cost_estimate` to see what it was doing, perhaps it was doing a doubly robust estimate.